bundles/org.openhab.binding.speedtest/src/main/java/org/openhab/binding/speedtest/internal/SpeedtestHandler.java

```diff
@@ -0,0 +1,538 @@
+    private String resultUrl = "";
+    private String server = "";
+
+    /**
+     * Contains information about which operating system openHAB is running on.
+     */
+    public enum OS {
+        WINDOWS,
+        LINUX,
+        MAC,
+        SOLARIS,
+        UNKNOWN,
+        NOT_SET
+    }
+
+    public SpeedtestHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("handleCommand channel: {} command: {}", channelUID, command);
+        String ch = channelUID.getId();
+        if (command instanceof RefreshType) {
+            if (!server.isBlank()) {
+                updateChannels();
+            }
+            return;
+        }
+        if (ch.equals(SpeedtestBindingConstants.TRIGGER_TEST)) {
+            if (command instanceof OnOffType) {
+                if (command == OnOffType.ON) {
+                    getSpeed();
+                    updateState(channelUID, OnOffType.OFF);
+                }
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(SpeedtestConfiguration.class);
+        pollingInterval = config.refreshInterval;
+        serverID = config.serverID;
+        if (!config.execPath.isEmpty()) {
+            speedTestCommand = config.execPath;
+        } else {
+            switch (getOperatingSystemType()) {
+                case WINDOWS:
+                    speedTestCommand = "";
+                    break;
+                case LINUX:
+                case MAC:
+                case SOLARIS:
+                    speedTestCommand = "/usr/bin/speedtest";
+                    break;
+                default:
+                    speedTestCommand = "";
+            }
+        }
+
+        updateStatus(ThingStatus.UNKNOWN);
+
+        if (!checkConfig(speedTestCommand)) { // check the config
+            return;
+        }
+        if (!getSpeedTestVersion()) {
+            return;
+        }
+        getServerList();
+        updateStatus(ThingStatus.ONLINE);
+        isRunning = true;
+        onUpdate(); // Setup the scheduler
+    }
+
+    /**
+     * This is called to start the refresh job and also to reset that refresh job when a config change is done.
+     */
+    private void onUpdate() {
+        logger.debug("Polling Interval Set: {}", pollingInterval);
+        if (pollingInterval > 0) {
+            ScheduledFuture<?> pollingJob = this.pollingJob;
+            if (pollingJob == null || pollingJob.isCancelled()) {
+                pollingJob = scheduler.scheduleWithFixedDelay(pollingRunnable, 0, pollingInterval, TimeUnit.MINUTES);
+            }
+        }
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Disposing Speedtest Handler Thing");
+        isRunning = false;
+        ScheduledFuture<?> pollingJob = this.pollingJob;
+        if (pollingJob != null && !pollingJob.isCancelled()) {
```

Comment:
No need to check if cancelled:
```suggestion
        if (pollingJob != null) {
```

Reply:
removed